api/net/Areas/Subscriber/Controllers/UserController.cs

```diff
@@ -0,0 +1,112 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+using Swashbuckle.AspNetCore.Annotations;
+using TNO.API.Areas.Services.Models.Minister;
+using TNO.API.CSS;
+using TNO.API.Models;
+using TNO.DAL.Services;
+using TNO.Entities;
+using TNO.Entities.Models;
+using TNO.Keycloak;
+
+namespace TNO.API.Areas.Subscriber.Controllers;
+
+/// <summary>
+/// UserController class, provides User endpoints for the admin api.
+/// </summary>
+[ClientRoleAuthorize(ClientRole.Subscriber)]
+[ApiController]
+[Area("subscriber")]
+[ApiVersion("1.0")]
+[Route("api/v{version:apiVersion}/[area]/users")]
+[Route("api/[area]/users")]
+[Route("v{version:apiVersion}/[area]/users")]
+[Route("[area]/users")]
+[ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.Unauthorized)]
+[ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.Forbidden)]
+public class UserController : ControllerBase
+{
+    #region Variables
+    private readonly IUserService _userService;
+    private readonly ICssHelper _cssHelper;
+    private readonly JsonSerializerOptions _serializerOptions;
+    #endregion
+
+    #region Constructors
+    /// <summary>
+    /// Creates a new instance of a UserController object, initializes with specified parameters.
+    /// </summary>
+    /// <param name="userService"></param>
+    /// <param name="serializerOptions"></param>
+    /// <param name="cssHelper"></param>
+    public UserController(IUserService userService, ICssHelper cssHelper, IOptions<JsonSerializerOptions> serializerOptions)
+    {
+        _userService = userService;
+        _cssHelper = cssHelper;
+        _serializerOptions = serializerOptions.Value;
+
+        
+    }
+    #endregion
+
+    #region Endpoints
+    /// <summary>
+    /// Find a page of user for the specified query filter.
+    /// </summary>
+    /// <returns></returns>
+    [HttpGet]
+    [Produces(MediaTypeNames.Application.Json)]
+    [ProducesResponseType(typeof(IPaged<UserModel>), (int)HttpStatusCode.OK)]
+    [SwaggerOperation(Tags = new[] { "User" })]
+    public IActionResult Find()
+    {
+        var uri = new Uri(this.Request.GetDisplayUrl());
+        var query = Microsoft.AspNetCore.WebUtilities.QueryHelpers.ParseQuery(uri.Query);
+        var result = _userService.Find(new DAL.Models.UserFilter(query));
+        var page = new Paged<UserModel>(result.Items.Select(ds => new UserModel(ds, _serializerOptions)), result.Page, result.Quantity, result.Total);
+        return new JsonResult(page);
+    }
+
+    /// <summary>
+    /// Find user for the specified 'id'.
+    /// </summary>
+    /// <param name="id"></param>
+    /// <returns></returns>
+    [HttpGet("{id}")]
+    [Produces(MediaTypeNames.Application.Json)]
+    [ProducesResponseType(typeof(UserModel), (int)HttpStatusCode.OK)]
+    [ProducesResponseType(typeof(string), (int)HttpStatusCode.NoContent)]
+    [SwaggerOperation(Tags = new[] { "User" })]
+    public IActionResult FindById(int id)
```

Comment:
Looking at your PR it appears you're using this endpoint to fetch the current user's info.  But you're opening the door for requesting every single user.  While it probably doesn't matter in our solution, this is essentially a security hole in nearly all other solutions.